packages/next/src/server/config.ts

```diff
@@ -1272,13 +1274,16 @@ export default async function loadConfig(
       userConfigModule.default || userConfigModule
     )) as NextConfig
 
-    if (!process.env.NEXT_MINIMAL) {
+    // Only validate the config against schema in:
+    // * non minimal mode
+    // * when `loadConfig` result is not silent
+    if (!process.env.NEXT_MINIMAL && validateConfig) {
```

Comment:
It sounds like we're fixing the wrong thing here, if validate is run twice it should pass both times unless we are mutating something we shouldn't be